### PR TITLE
control/controlclient: delete old naclbox code, require ts2021 Noise

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #11585
Updates https://github.com/tailscale/corp/issues/18882

### CODE REVIEW

**Comment by bradfitz** [comment]
(I think integration tests will fail due to the mock control server only doing naclbox still... need to fix that first in this PR)

File: control/controlclient/direct.go [hunk 1]
```diff
@@ -494,20 +488,22 @@ func (c *Direct) doLogin(ctx context.Context, opt loginOpt) (mustRegen bool, new
...
 		c.mu.Lock()
-		c.serverKey = keys.LegacyPublicKey
+		c.serverLegacyKey = keys.LegacyPublicKey
 		c.serverNoiseKey = keys.PublicKey
 		c.mu.Unlock()
 		serverKey = keys.LegacyPublicKey
 		serverNoiseKey = keys.PublicKey
 
-		// For servers supporting the Noise transport,
 		// proactively shut down our TLS TCP connection.
```
Comment (maisem) [inline control/controlclient/direct.go:497]: Proactively

File: control/controlclient/direct.go [hunk 2]
```diff
@@ -1224,26 +1189,10 @@ func (c *Direct) decodeMsg(msg []byte, v any, mkey key.MachinePrivate) error {
...
-		return fmt.Errorf("response: %v", err)
-	}
-	return nil
 }
 
-// encode JSON encodes v. If serverNoiseKey is not specified, it uses the serverKey and mkey to
-// seal the message into a NaCl-crypto-box.
-func encode(v any, serverKey, serverNoiseKey key.MachinePublic, mkey key.MachinePrivate) ([]byte, error) {
+// encode JSON encodes v.
+func encode(v any, serverNoiseKey key.MachinePublic) ([]byte, error) {
```
Comment (maisem) [inline control/controlclient/direct.go:1195]: can drop `serverNoiseKey`

**Review by maisem — APPROVED** [review]

File: control/controlclient/direct.go [hunk 3]
```diff
@@ -614,21 +610,16 @@ func (c *Direct) doLogin(ctx context.Context, opt loginOpt) (mustRegen bool, new
...
-		url = strings.Replace(url, "http:", "https:", 1)
+
+	request.Version = tailcfg.CurrentCapabilityVersion
+	httpc, err := c.getNoiseClient()
+	if err != nil {
+		return regen, opt.URL, nil, fmt.Errorf("getNoiseClient: %w", err)
 	}
-	bodyData, err := encode(request, serverKey, serverNoiseKey, machinePrivKey)
+	url := fmt.Sprintf("%s/machine/register", c.serverURL)
+	url = strings.Replace(url, "http:", "https:", 1)
```
Comment (zhangguanzhang) [inline control/controlclient/direct.go:620]: this will always request to http, This will timeout when the tailscale client connects to an http headscale

**Pull request merged**